test/spec/grammar/dsl_api_test.js

```diff
@@ -0,0 +1,140 @@
+        expect(ast.entities[0]).to.deep.eql({
+          name: 'A',
+          tableName: 'A',
+          body: [
+            {
+              name: 'field',
+              type: 'String',
+              validations: [],
+              javadoc: null
+            }
+          ],
+          javadoc: null
+        });
+      });
+    });
+
+    describe('with a lexing error', () => {
+      it('throws an error', () => {
+        const invalidInput = 'entity ± {';
+        expect(() => parse(invalidInput)).to
+          .throw('unexpected character: ->±<- at offset: 7, skipped 1 characters.');
+      });
+    });
+
+    describe('with a Parsing error', () => {
+      describe('with a MismatchedTokenException', () => {
+        it('throws an error', () => {
+        // wrong type of closing parenthesis ']'
+          const invalidInput = 'entity Person { ]';
+          expect(() => parse(invalidInput)).to
+            .throw('MismatchedTokenException: Expecting --> \'}\' <-- but found --> \']\' <--\n' +
+            '\tat line: 1, column: 17');
+        });
+      });
+
+      describe('with a MismatchedTokenException for EOF', () => {
+        it('throws an error', () => {
+          // No Closing parenthesis
+          const invalidInput = 'entity Person {';
+          expect(() => parse(invalidInput)).to
+            .throw('MismatchedTokenException: Expecting --> \'}\' <-- but found --> \'\' <--');
+        });
+      });
+    });
+
+    describe('with a semantic validation error', () => {
+      it('throws an error', () => {
+        // lower case entityName first char
+        const invalidInput = 'entity person { }';
+        expect(() => parse(invalidInput)).to
+          .throw('Entity name must match: /^[A-Z][A-z0-9]*$/\n' +
+            '\tat line: 1, column: 8');
+      });
+    });
+  });
+
+  describe('when wanting an auto-completion', () => {
+    describe('with an empty text', () => {
+      it('provides suggestions', () => {
+        const input = '';
+        const result = getSyntacticAutoCompleteSuggestions(input);
+        expect(result).to.have.lengthOf(17);
+        expect(result).to.have.members([
+          tokens.APPLICATION,
+          tokens.NAME,
+          tokens.ENTITY,
+          tokens.RELATIONSHIP,
+          tokens.ENUM,
+          tokens.DTO,
+          tokens.SERVICE,
+          tokens.SEARCH,
+          tokens.MICROSERVICE,
+          tokens.COMMENT,
+          tokens.PAGINATE,
+          tokens.SKIP_CLIENT,
+          tokens.SKIP_SERVER,
+          tokens.NO_FLUENT_METHOD,
+          tokens.ANGULAR_SUFFIX,
+          tokens.FILTER,
+          tokens.CLIENT_ROOT_FOLDER
+        ]);
+      });
+    });
+    describe('with a custom start rule', () => {
+      it('provides suggestions', () => {
+        const input = 'lastName string ';
+        const result = getSyntacticAutoCompleteSuggestions(
+          input,
+          'fieldDeclaration'
+        );
+        expect(result).to.have.lengthOf(4);
```

Comment:
Same comment

Reply:
That is a good suggestion, but that is not part of code of this commit. (renamed file)